operate/importer/src/main/java/io/camunda/operate/zeebeimport/RecordsReaderHolder.java

```diff
@@ -75,9 +79,32 @@ public boolean hasPartitionCompletedImporting(final int partitionId) {
     return partitionsCompletedImporting.contains(partitionId);
   }
 
-  public int incrementEmptyBatches(final int partitionId, final ImportValueType importValueType) {
+  public void incrementEmptyBatches(final int partitionId, final ImportValueType importValueType) {
     final var reader = getRecordsReader(partitionId, importValueType);
-    return countEmptyBatchesAfterImportingDone.merge(reader, 1, Integer::sum);
+    countEmptyBatchesAfterImportingDone.merge(reader, 1, Integer::sum);
+  }
+
+  public boolean isRecordReaderCompletedImporting(
+      final int partitionId, final ImportValueType importValueType) {
+    if (hasPartitionCompletedImporting(partitionId)) {
+
+      final var reader = getRecordsReader(partitionId, importValueType);
+      return Objects.equals(
+          countEmptyBatchesAfterImportingDone.get(reader),
+          MINIMUM_EMPTY_BATCHES_FOR_COMPLETED_READER);
+    }
```

Comment:
Good catch, this was a bug that would have cause some issues in conjunction with the bug I fixed in https://github.com/camunda/camunda/pull/24849/commits/01a2744621b8e07479605bfa02d94ed149da2161. If both bugs were left in. After a record reader is marked as completed, the next subsequent 8.7.0 record would cause it to be marked as not completed 👍